### PR TITLE
Clean up the debug output for unit animations

### PR DESCRIPTION
**Pull request by CelticMinstrel**
Incorporates (and thus supercedes) PR #705

### CODE REVIEW

File: src/units/animation.cpp [hunk 1]
```diff
@@ -1096,118 +1096,78 @@ std::string unit_animation::debug() const
...
-				case (unit_animation::HIT)     : std::cout << "hit"; break;
-				case (unit_animation::MISS)    : std::cout << "miss"; break;
-				case (unit_animation::KILL)    : std::cout << "kill"; break;
-				case (unit_animation::INVALID) : std::cout << "invalid"; break;
-			}
-		}
-		std::cout << '\n';
+		std::vector<std::string> hits;
+		std::transform(u_animation.hits_.begin(), u_animation.hits_.end(), std::back_inserter(hits), unit_animation::hit_type::enum_to_string);
+		outstream << "\thits=" << utils::join(hits) << '\n';
```
Comment (Vultraz) [inline src/units/animation.cpp]: What does `\t` do?

File: src/units/animation.cpp [hunk 2]
```diff
@@ -1096,118 +1096,78 @@ std::string unit_animation::debug() const
...
-				case (map_location::SOUTH)     : std::cout << "s"; break;
-				case (map_location::SOUTH_WEST): std::cout << "sw"; break;
-				case (map_location::NORTH_WEST): std::cout << "nw"; break;
-				default: break;
-			}
-		}
-		std::cout << '\n';
+		std::vector<std::string> dirs;
+		std::transform(u_animation.directions_.begin(), u_animation.directions_.end(), std::back_inserter(dirs), map_location::write_direction);
+		outstream << "\tdirections=" << utils::join(dirs) << '\n';
```
Comment (Vultraz) [inline src/units/animation.cpp]: As above.

File: src/units/animation.cpp [hunk 3]
```diff
@@ -1096,118 +1096,78 @@ std::string unit_animation::debug() const
...
 	}
 	if (u_animation.terrain_types_.size() > 0) {
-		i=0;
-		std::cout << "\tterrain=";
-		for (const t_translation::t_terrain terrain : u_animation.terrain_types_) {
-			if (i>0) std::cout << ','; i++;
-			std::cout << terrain;
-		}
-		std::cout << '\n';
+		outstream << "\tterrain=" << utils::join(u_animation.terrain_types_) << '\n';
```
Comment (Vultraz) [inline src/units/animation.cpp]: As above.

File: src/units/animation.cpp (hunk 1 above)
Comment (Wedge009) [inline src/units/animation.cpp]: Doesn't it output a tab space?

Comment (AI0867) [inline src/units/animation.cpp]: Yes.

File: src/units/animation.hpp [hunk 4]
```diff
@@ -29,14 +30,19 @@ class unit_animation
...
+			(KILL, "kill")
+			(INVALID, "invalid")
+		);
 
 		static const std::vector<std::string>& all_tag_names();
 		static void fill_initial_animations( std::vector<unit_animation> & animations, const config & cfg);
 		static void add_anims( std::vector<unit_animation> & animations, const config & cfg);
 
-		int matches(const display &disp,const map_location& loc,const map_location& second_loc,const unit* my_unit,const std::string & event="",const int value=0,hit_type hit=INVALID,const attack_type* attack=nullptr,const attack_type* second_attack = nullptr, int value2 =0) const;
+	int matches(const display &disp,const map_location& loc,const map_location& second_loc,const unit* my_unit,const std::string & event="",const int value=0,hit_type hit=hit_type::INVALID,const attack_type* attack=nullptr,const attack_type* second_attack = nullptr, int value2 =0) const;
```
Comment (aginor) [inline src/units/animation.hpp]: you've broken the indentation here.

Comment (CelticMinstrel) [inline src/units/animation.hpp]: Ugh, that would be because I have XCode set to re-indent when I type :, and this file is indented weirdly.

**Comment by Vultraz** [comment]
Are we merging this for 1.13.5?

**Comment by aginor** [comment]
That would go against a "feature freeze". It is strictly speaking not needed (afaik, there's no way to actually run the code) but it may nevertheless introduce risk.

**Comment by Vultraz** [comment]
Ok, I'll postpone it.

**Pull request merged**